x-pack/legacy/plugins/lens/public/indexpattern_plugin/loader.ts

```diff
@@ -50,10 +50,12 @@ export const getIndexPatterns = (chrome: Chrome, toastNotifications: ToastNotifi
           id,
           type,
           title: attributes.title,
-          fields: (JSON.parse(attributes.fields) as IndexPatternField[]).filter(
-            ({ type: fieldType, esTypes }) =>
-              fieldType !== 'string' || (esTypes && esTypes.includes('keyword'))
-          ),
+          fields: (JSON.parse(attributes.fields) as IndexPatternField[])
+            .map(field => ({ ...field, indexPatternId: id }))
```

Comment:
I don't like this approach- it brings us _even further_ from the index pattern service in terms of implementation, and will make it harder to transition in the future. My preferred solution is to attach more metadata on the drag state.

Reply:
It's fine for me, I don't have a real preference for either solution. We have to map over the fields anyway to add the restrictions, so I don't think a step like this will ever become obsolete.

One thing I like about adding it only to the dragged state is that our core data structure becomes less redundant, but that comes with an additional type one has to handle.

While looking into that, I noticed that we have a similar situation with the column data type - it also includes an indexpattern property even though that's implied by the layer. I'll also remove it there.